RecoEgamma/EgammaElectronProducers/plugins/GsfElectronProducer.cc

```diff
@@ -91,3 +113,138 @@ void GsfElectronProducer::checkPfTranslatorParameters( edm::ParameterSet const &
      }
    }
  }
+
+
+// now deprecated
```

Comment:
(This comment was not originally written by you but) why to have this method still called if deprecated?

Reply:
That's a good question. It would be very valuable to get some history lesson from @Sam-Harper, but what I understood is that from the POV of the `GsfElectronProducer` it's not deprecated since it's used in this producer. My guess is that the `GEDGsfElectronProducer` was supposed to superseed the `GsfElectronProducer`, and therefore all functions in `GsfElectronAlgo` which are only used by the `GsfElectronProducer` got flagged as deprecated. With the function being at the new location, this comment should maybe just get removed.

Anyway, tha's just my interpretation after reading the code...

Reply:
I left this comment in with my clang-format commit for now.

Reply:
Hi @guitargeek , I think you are correct on this. This was part of the great Josh/Lindsey rewrite and changed how we handle e/gamma & particle flow. So this was the old way and GEDGsfElectrons which are really the thing we use now are the new way